mlflow/pyfunc/__init__.py

```diff
@@ -705,6 +707,16 @@ def _validate_prediction_input(
             )
         return data, params
 
+    def _update_dependencies_schema_in_prediction_context(self):
+        if self._model_meta and self._model_meta.metadata and (context := get_prediction_context()):
+            dependencies_schema = self._model_meta.metadata.get("dependencies_schemas", {})
```

Comment:
I see

https://github.com/mlflow/mlflow/blob/12898b30d5a0b55fccd62a2ef9a11b72fa45ec53/mlflow/models/dependencies_schema.py#L200

Reply:
@sunishsheth2009 Can we only use `dependencies_schema`, not `dependencies_schemas`?

Reply:
I agree, its error prone.. We should surely unify to `dependencies_schemas` since it would include multiple dependencies and multiple schemas for the same dependency as well. 
Maybe we can update the context? Sorry for not unifying before. My bad

Reply:
Filed https://github.com/mlflow/mlflow/pull/12079